src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>A <see cref="unit" /> <see cref="task" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val iteriAsync: action: (int -> 'T -> #Task<unit>) -> source: taskSeq<'T> -> Task<unit>
 
-    /// Maps over the taskSeq, applying the mapper function to each item. This function is non-blocking.
-    val map: mapper: ('T -> 'U) -> source: taskSeq<'T> -> taskSeq<'U>
-
     /// <summary>
     /// Builds a new task sequence whose elements are the corresponding elements of the input task
     /// sequence <paramref name="source" /> paired with the integer index (from 0) of each element.
     /// Does not evaluate the input sequence until requested.
     /// </summary>
+    ///
     /// <param name="source">The input task sequence.</param>
     /// <returns>The resulting task sequence of tuples.</returns>
-    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     val indexed: source: taskSeq<'T> -> taskSeq<int * 'T>
 
-    /// Maps over the taskSeq with an index, applying the mapper function to each item. This function is non-blocking.
+    /// <summary>
+    /// Builds a new task sequence whose elements are the results of applying the <paramref name="action" />
+    /// function to each of the elements of the input task sequence in <paramref name="source" />.
+    /// The given function will be applied as elements are demanded using the <see cref="MoveNextAsync" />
+    /// method on async enumerators retrieved from the input task sequence.
+    /// Does not evaluate the input sequence until requested.
+    /// </summary>
+    ///
+    /// <param name="mapping">A function to transform items from the input task sequence.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    val map: mapper: ('T -> 'U) -> source: taskSeq<'T> -> taskSeq<'U>
+
+    /// <summary>
+    /// Builds a new task sequence whose elements are the results of applying the <paramref name="action" />
+    /// function to each of the elements of the input task sequence in <paramref name="source" />, passing
+    /// an extra index argument to the <paramref name="action" /> function.
+    /// The given function will be applied as elements are demanded using the <see cref="MoveNextAsync" />
+    /// method on async enumerators retrieved from the input task sequence.
+    /// Does not evaluate the input sequence until requested.
+    /// </summary>
+    ///
+    /// <param name="mapping">A function to transform items from the input task sequence that also access the current index.</param>
```

Comment:
```suggestion
    /// <param name="mapping">A function to transform items from the input task sequence. The logical index within the sequence is supplied alongside the item.</param>
```

Reply:
This is done for parity with seq. We should improve the docs there before doing it here.